src/RadioButton/RadioButton.js

```diff
@@ -132,26 +127,11 @@ const RadioButton = React.createClass({
     };
   },
 
-  getInitialState() {
-    return {
-      muiTheme: this.context.muiTheme || getMuiTheme(),
-    };
-  },
-
-  getChildContext() {
-    return {
-      muiTheme: this.state.muiTheme,
-    };
-  },
-
-  componentWillReceiveProps(nextProps, nextContext) {
-    this.setState({
-      muiTheme: nextContext.muiTheme || this.state.muiTheme,
-    });
+  getTheme() {
+    return this.context.muiTheme.radioButton;
   },
 
-  getTheme() {
-    return this.state.muiTheme.radioButton;
+  handleStateChange() {
```

Comment:
It's for a required prop in EnhancedSwitch. @oliviertassinari has removed it in #3408.